projects/filigran-website/components/display-all-icons.tsx

```diff
@@ -1,16 +1,48 @@
+'use client'
 import * as FiligranIcon from 'filigran-icon'
-import {Fragment} from 'react'
+import React, {Fragment,ReactElement,useState,useMemo} from 'react'
+import Popup from './icon-popup';
+import SearchBar from './icon-searchbar';
 
 export const DisplayAllIcons = () => {
+  const [selectedIcon, setSelectedIcon] = useState<ReactElement | null>(null);
+  const [selectedIconLabel, setSelectedIconLabel] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const handleIconClick = (Icon : React.ComponentType<any> ,icon : string) => {
+    setSelectedIcon(<Icon className="w-20 h-20"/>);
+    setSelectedIconLabel(icon);
+    setIsOpen(true);
+  }
+
+  const handleClose = () => {
+  setIsOpen(false);
+  setSelectedIcon(null);
+  setSelectedIconLabel('');
+  } 
+
   const allIcons = Object.keys(FiligranIcon).filter((key) => key !== 'default')
+
+  const filteredIcons = useMemo(() => {
+    return allIcons.filter(icon => 
+      icon.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+  }, [allIcons, searchTerm]);
   const getIcon = (icon: string) => {
     // @ts-ignore
-    const Icon = FiligranIcon[icon]
-    return <Icon className="h-16 w-16" />
+    const Icon = FiligranIcon[icon]  as React.ComponentType<any>;
+    return <Icon className="h-16 w-16 hover:border-[1px] border-blue rounded-lg"  onClick={() => handleIconClick(Icon,icon)} />
   }
   return (
+    <div className='container'>
```

Comment:
not sure that "container" is working anymore. I remove the [tailwindcss-typography](https://github.com/tailwindlabs/tailwindcss-typography) plugin I want to use our own theme now.
But in case that it's not really use, you can wrap with `<></>`